File: backend/app/api/projects.py
from flask import Blueprint, jsonify

bp = Blueprint('projects', __name__, url_prefix='/api/projects')

@bp.route('/')
def get_projects():
    projects = [
        {
            'id': 1,
            'title': 'LLM Project',
            'description': 'AI stack project using langchain, chromadb, and Ollama',
            'tags': ['Python', 'AI', 'LLM'],
            'github': 'https://github.com/yourusername/llm-project'
        },
        {
            'id': 2,
            'title': 'Portfolio Website',
            'description': 'This portfolio website built with React and Flask',
            'tags': ['React', 'Flask', 'CSS'],
            'github': 'https://github.com/yourusername/portfolio'
        }
    ]
    
    return jsonify(projects)

@bp.route('/<int:project_id>')
def get_project(project_id):
    # In a real app, this would fetch from a database
    projects = {
        1: {
            'id': 1,
            'title': 'LLM Project',
            'description': 'AI stack project using langchain, chromadb, and Ollama',
            'long_description': 'A detailed exploration of building LLM applications with Python...',
            'tags': ['Python', 'AI', 'LLM'],
            'github': 'https://github.com/yourusername/llm-project',
            'image': 'llm_project.jpg'
        },
        2: {
            'id': 2,
            'title': 'Portfolio Website',
            'description': 'This portfolio website built with React and Flask',
            'long_description': 'My personal portfolio site built with React and Flask...',
            'tags': ['React', 'Flask', 'CSS'],
            'github': 'https://github.com/yourusername/portfolio',
            'image': 'portfolio.jpg'
        }
    }
    
    project = projects.get(project_id)
    if not project:
        return jsonify({'error': 'Project not found'}), 404
        
    return jsonify(project)

File: frontend/src/pages/Projects.js
import React from 'react';
import '../styles/Projects.css';

function Projects() {
  const projects = [
    {
      id: 1,
      title: "Biomedical Chat Assistant",
      description: "AI stack project using langchain, chromadb, and Ollama",
      tags: ["Python", "AI", "LLM"],
      github: "https://github.com/sadieflick/biomedical_graphrag"
    },
    {
      id: 2,
      title: "Portfolio Website",
      description: "This portfolio website built with React and Flask",
      tags: ["React", "Flask", "CSS"],
      github: "https://github.com/sadieflick/sadie-portfolio"
    }
  ];

  return (
    <div className="projects-container">
      <h1>My Projects</h1>
      <div className="projects-grid">
        {projects.map(project => (
          <div className="project-card" key={project.id}>
            <h2>{project.title}</h2>
            <p>{project.description}</p>
            <div className="tags">
              {project.tags.map(tag => (
                <span className="tag" key={tag}>{tag}</span>
              ))}
            </div>
            <a href={project.github} target="_blank" rel="noopener noreferrer">View on GitHub</a>
          </div>
        ))}
      </div>
    </div>
  );
}

export default Projects;
